atari-DQN/replay_memory.py

```diff
@@ -97,13 +111,15 @@ def encode_states(self, idxs: np.ndarray) -> (torch.tensor, torch.tensor):
         :return: minibatches of state and successor states
         """
 
-        # Initialize current and next states
-        cur_states = torch.zeros(((len(idxs), self.history) + self._obs_shape),
+        # Initialize current and next states tensors (pre zero-padded)
+        _state_tensor_shape = ((len(idxs), self.history) + self._obs_shape[-2:])
```

Comment:
Not taking into account the possibility that each observation can have multiple channels (in addition to multiple observations needed to be stacked together to form a single state. I think it should be:
```python
_state_tensor_shape = ((len(idxs), self.history*self._obs_shape[0]) + self._obs_shape[-2:])
```

Reply:
Note: this is fixed in the RL-767 SMiRL repository